### DISCUSSION TITLE
Suppressing Auto Instrumentation Conditionally

### DISCUSSION THREAD

**Discussion by hcelaloner** [open-telemetry/opentelemetry-java#6887] [Q&A] [answered]
In some conditions, we want to disable auto-instrumentation for an execution path. We mainly aim to reduce the amount of spans generated by some execution flows by aggregating/batching them into one span.  Let me explain it in a small snippet. 

For instance, in the following snippet, we have the 'getPricesByIds' function which can be used to fetch 50 price documents from Couchbase via 50 database calls. In this case, auto instrumentation produces 50 child spans for each Couchbase call. Is there a way in which I can generate only one span for this operation instead of 50 without disabling the whole Couchbase instrumentation? I don't know but is there some sort of suppress signal we can use in `getPricesByIds` function to prevent any child span creation in the subsequent calls? However, I don't want to disable it for  `getPricesByIds` function.
 
```java
public List<Price> getPricesByIds(List<String> ids) {
  return ids.stream().map(priceRepository::getByID).collect(Collectors.toList());
}
  
public Price getPriceById(String id) {
  return priceRepository.getByID(id);
}
```

**Comment by trask** [comment, marked as answer]
check out https://github.com/open-telemetry/opentelemetry-specification/issues/530

as a short-term solution, if you are using (the default) parent-based sampler, I think you can do

```
public List<Price> getPricesByIds(List<String> ids) {
  SpanContext current = Span.current().getSpanContext();
  SpanContext suppressing = SpanContext.create(current.getTraceId(), current.getSpanId(), TraceFlags.getDefault(), current.getTraceState());
  try (Scope scope = Span.wrap(suppressing).makeCurrent()) {
    return ids.stream().map(priceRepository::getByID).collect(Collectors.toList());
  }
}
```

the key above is `TraceFlags.getDefault()` is non-recording (compared to `TraceFlags.getSampled()`

**Reply by hcelaloner** [reply]
Thank you for the solution. It worked 🙏
